classes/submission/reviewAssignment/Repository.php

```diff
@@ -227,6 +227,28 @@ public function deleteMany(Collector $collector)
         }
     }
 
+    /**
+     * Delete all review assignments for a given context ID.
+     */
+    public function deleteByContextId(int $contextId): void
+    {
+        // using reviewAssignmentCollector to fetch ids of all review assignments for the context
+        $reviewAssignmentCollector = $this->getCollector();
+        $reviewAssignmentCollector->filterByContextIds([$contextId]);
+        $reviewAssignmentIds = $reviewAssignmentCollector->getIds();
+    
+        foreach ($reviewAssignmentIds as $reviewAssignmentId) {
+            $reviewAssignment = $this->get($reviewAssignmentId);
```

Comment:
You should be able to use `$this->dao->deleteById($reviewAssignmentId)` rather than needing to fetch/instantiate the object.

Reply:
@asmecher Thank you for comment regarding the use of `$this->dao->deleteById($reviewAssignmentId)`. But I noticed that our current `DAO.php` doesn't have this method implemented yet. Would you prefer I go ahead and add it?

Reply:
Have a look at the parent class: https://github.com/pkp/pkp-lib/blob/3a92240297339f296ae561c7e3b291b156957eb8/classes/db/SchemaDAO.php#L211